desktop/core/src/desktop/js/utils/hooks/useQueueProcessor.test.tsx

```diff
@@ -21,108 +21,106 @@ describe('useQueueProcessor', () => {
+    const { result } = renderHook(() =>
+      useQueueProcessor(mockProcessItem, { concurrentProcess: CONCURRENT_PROCESS })
+    );
 
+    expect(result.current.queue).toEqual([]);
     act(() => {
       result.current.enqueue(['item1', 'item2', 'item3', 'item4']);
     });
 
-    expect(result.current.pendingQueue).toEqual(['item3', 'item4']);
-    expect(result.current.processingQueue.length).toEqual(CONCURRENT_PROCESS);
-    expect(result.current.processingQueue).toEqual(['item1', 'item2']);
-    expect(result.current.completedQueue).toEqual([]);
+    expect(result.current.queue).toEqual(['item3', 'item4']);
+    expect(result.current.isLoading).toBe(true);
 
-    await waitFor(() => expect(result.current.completedQueue).toEqual(['item1', 'item2']));
-
-    expect(result.current.pendingQueue).toEqual([]);
-    expect(result.current.processingQueue.length).toEqual(CONCURRENT_PROCESS);
-    expect(result.current.processingQueue).toEqual(['item3', 'item4']);
-    expect(result.current.completedQueue).toEqual(['item1', 'item2']);
+    await waitFor(() => expect(result.current.queue).toEqual([]));
+    await waitFor(() => expect(result.current.isLoading).toBe(false));
   });
 
-  it('should remove items from pending queue when dequeued', async () => {
-    const { result } = renderHook(() => useQueueProcessor(mockProcessItem));
+  it('should add items to the queue when enqueued', () => {
+    const { result } = renderHook(() =>
+      useQueueProcessor(mockProcessItem, { concurrentProcess: 1 })
+    );
+
+    expect(result.current.queue).toEqual([]);
 
     act(() => {
-      result.current.enqueue(['item1', 'item2', 'item3']);
+      result.current.enqueue(['item1', 'item2']);
     });
 
-    expect(result.current.processingQueue).toEqual(['item1']);
-    expect(result.current.pendingQueue).toEqual(['item2', 'item3']);
+    expect(result.current.queue).toEqual(['item2']);
 
     act(() => {
-      result.current.dequeue('item3');
+      result.current.enqueue(['item3']);
     });
 
-    expect(result.current.pendingQueue).toEqual(['item2']);
-    expect(result.current.processingQueue).toEqual(['item1']);
+    expect(result.current.queue).toEqual(['item2', 'item3']);
   });
 
-  it('should remove items from pending queue when dequeued', async () => {
-    const { result } = renderHook(() => useQueueProcessor(mockProcessItem));
+  it('should remove items from the queue when dequeued', async () => {
+    const { result } = renderHook(() =>
+      useQueueProcessor(mockProcessItem, { concurrentProcess: 1 })
+    );
 
     act(() => {
       result.current.enqueue(['item1', 'item2', 'item3']);
     });
 
-    expect(result.current.processingQueue).toEqual(['item1']);
-    expect(result.current.pendingQueue).toEqual(['item2', 'item3']);
+    expect(result.current.queue).toEqual(['item2', 'item3']);
+    expect(result.current.isLoading).toBe(true);
 
     act(() => {
-      result.current.dequeue('item3');
+      result.current.dequeue('item2');
```

Comment:
I'm being picky here, but If you would use objects like this

```
result.current.enqueue([{name:'item1'}, {name:'item2'}, {name:'item3'}]);
result.current.dequeue({name:'item2'});
```
the test would likely fail since you are comparing on reference equality.